### PR TITLE
BLOCKCLONE backup method

### PR DESCRIPTION
**Pull request by gozora**
This commit implements block device cloning using `dd` and `ntfsclone`.

Documentation is work in progress. Current documentation draft can be found [here](https://gozora.sk/bc.html). I'll add documentation to upstream ReaR code soon.
(sorry lot of work lately :-( )

V.

### CODE REVIEW

**Comment by gozora** [comment]
Would you mind if I create separate section  (12-BLOCKCLONE.adoc) in _doc/user-guide_ ?

**Comment by jsmeix** [comment]
I would even recommend to use separated documentation files
for separated backup methods - in particular in this case
because BLOCKCLONE is totally different than other
backup methods.

**Comment by gozora** [comment]
Thanks @jsmeix ,

That's exactly what I thought. 
I'll add one simple example to _04-scenarios.adoc_ with reference to  _12-BLOCKCLONE.adoc_, for more details.

V.

**Comment by jsmeix** [comment]
@gozora 
right now I found
<pre>
if [[ "$BLOCKCLONE_STRICT_PARTITIONING" =~ ^[yY1] ]]
</pre>
It posible I would recommend to use the existing
is_true and is_false functions, see
"Relax-and-Recover functions" in
https://github.com/rear/rear/wiki/Coding-Style

**Comment by jsmeix** [comment]
@gozora 
I found
<pre>
case "$label" in
    "msdos")
</pre>
Better use paired parenthesis for case patterns
so that editor commands (like '%' in 'vi') that
check for matching opening and closing parenthesis
work everywhere in the code, see
"Paired parenthesis" in
https://github.com/rear/rear/wiki/Coding-Style

**Comment by jsmeix** [comment]
@gozora 
it seems I am a bit slow on the uptake - could
you explain me how the change_default is menat
to be used - I read the comment but I do not understand
how setting $1 to a user input plus evaluating that
cold be useful functionality?

**Comment by gozora** [comment]
Hello @jsmeix,
I was looking for simmilar functionality as C's "pass by reference." Hence to change value of parameter passed to function.
I'm currently off my PC and typing on my phone is a bit pain. I'll try to explain it better tomorow.

V.

**Comment by gozora** [comment]
Hello @jsmeix,

_change_default_ function was meant as a simplification for dialog, where user is prompted for for either restore destination or destination for disk where partitions should be created:

A small extract from code:
```
echo -n "Restore $archive_name to device: [$BLOCKCLONE_SOURCE_DEV] "
change_default BLOCKCLONE_SOURCE_DEV
```

In this case value of BLOCKCLONE_SOURCE_DEV is loaded from *.conf file and is used as default. So user just hits _enter_ this value is used.
If user however enters some string, value in BLOCKCLONE_SOURCE_DEV will be replaced by user choice.

Hope I've explained it better this time.

V.

**Comment by gdha** [comment]
@gozora Thanks for the new feature  - tomorrow will do a new validation round to see whether it does not break something in other flows.

File: usr/share/rear/output/default/400_copy_disk_struct_files.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+../../backup/BLOCKCLONE/default/400_copy_disk_struct_files.sh
```
Comment (pcahyna) [inline usr/share/rear/output/default/400_copy_disk_struct_files.sh:1]: Hi @gozora , I wonder why are you linking the script from `backup` here under `output`. The script uses `backup_path`, which is not even mounted in the output stage. Also, why to link it under `default` and not under `BLOCKCLONE`? Or am I missing something?

**Pull request merged**